pkg/client/cache/delta_fifo.go

```diff
@@ -199,6 +258,7 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 		return KeyError{obj, err}
 	}
 	newDeltas := append(f.items[id], Delta{actionType, obj})
```

Comment:
I know its pre-existing, but can you add a comment indicating that this non-deterministic append (may or may not update `f.items[id]` is safe because access is locked and if `f.items[id]` is modified, it will be stomped before the function finishes.